dashboard/legacy/middleware/files_api.rb

```diff
@@ -413,13 +415,21 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
     end
 
     # Block libraries with PII/profanity from being published.
+    # Block main.json file from aichat lab flagged with profanity from being saved.
     #
     # Javalab's "backpack" feature uses libraries to allow students to share code
     # between their own projects -- skip this check for .java files, since in this use case
     # the files are only being used by a single user.
-    if endpoint == 'libraries' && file_type != '.java'
+    if (endpoint == 'libraries' && file_type != '.java') || LABS_TO_CHECK_FOR_PROFANITY.include?(app_type)
       begin
-        share_failure = ShareFiltering.find_failure(body, request.locale)
+        if app_type == 'aichat'
+          source = JSON.parse(body)['source']
+          source_json = JSON.parse(source)
+          text_to_check = source_json['systemPrompt'] + ' ' + source_json['retrievalContexts'].join(' ')
+          share_failure = ShareFiltering.find_failure(text_to_check, request.locale)
```

Comment:
Maybe we can wait till Sam gets back to add other fields. In the [jira ticket](https://codedotorg.atlassian.net/browse/LABS-791), she specifically requested that the system prompt and retrievals be filtered - I think bc these fields are sent to Sagemaker while model card info is not. With the tendency of WebPurify to have false positives, we can always follow-up with updating so that entire ` body` is checked.